packages/terra-functional-testing/src/services/wdio-visual-regression-service/methods/LocalCompare.js

```diff
@@ -8,73 +8,89 @@ import BaseCompare from './BaseCompare';
 
-      const compareData = await this.compareImages(referencePath, captured, ignoreComparison);
+      const ignoreComparison = _.get(context, 'options.ignoreComparison', this.ignoreComparison);
+      const compareData = await this.compareImages(referencePath, latestScreenshot, ignoreComparison);
 
       const { isSameDimensions } = compareData;
       const misMatchPercentage = Number(compareData.misMatchPercentage);
       const misMatchTolerance = _.get(context, 'options.misMatchTolerance', this.misMatchTolerance);
 
-      const diffPath = this.getDiffFile(context);
-
-      if (misMatchPercentage > misMatchTolerance) {
+      const isWithinMisMatchTolerance = misMatchPercentage <= misMatchTolerance;
+      if (!isWithinMisMatchTolerance || !isSameDimensions) {
         log.info(`Image is different! ${misMatchPercentage}%`);
         const png = compareData.getDiffImage().pack();
         await this.writeDiff(png, diffPath);
-
-        return this.createResultReport(referenceExists, misMatchPercentage, false, isSameDimensions);
-      } else { // eslint-disable-line no-else-return
+      } else {
         log.info('Image is within tolerance or the same');
+        // remove diff screenshot if it existed from a previous run
         await fs.remove(diffPath);
-
-        return this.createResultReport(referenceExists, misMatchPercentage, true, isSameDimensions);
       }
-    } else { // eslint-disable-line no-else-return
-      log.info('first run - create reference file');
-      await fs.outputFile(referencePath, base64Screenshot, 'base64');
-      return this.createResultReport(referenceExists);
+
+      return this.createResultReport(referenceExists, misMatchPercentage, isWithinMisMatchTolerance, isSameDimensions);
     }
+
+    log.info('first run - create reference file');
+    await fs.outputFile(referencePath, base64Screenshot, 'base64');
+    return this.createResultReport(referenceExists);
   }
 
   /**
-   * Compares two images with resemble
-   * @param  {Buffer|string} reference path to reference file or buffer
-   * @param  {Buffer|string} screenshot path to file or buffer to compare with reference
-   * @return {{misMatchPercentage: Number, isSameDimensions:Boolean, getImageDataUrl: function}}
+   * Compares the latest image to the latest image with resemble to determine if they are the same.
+   *
+   * @param {Buffer|string} reference - Path to reference file or buffer for the reference image.
+   * @param {Buffer|string} latest - Path to file or buffer of the latest image.
+   * @param {String} ignoreComparison - The image comparison algorithm to use.
+   * @returns {Object} - The screenshot comparison results returned as { misMatchPercentage: Number, isSameDimensions: Boolean, getImageDataUrl: function }.
    */
   // eslint-disable-next-line class-methods-use-this
   async compareImages(reference, screenshot, ignore = '') {
```

Comment:
Should `screenshot` be renamed to `latest` and `ignore` to `ignoreComparison` as they don't match the function documentation?

Reply:
yes, I must have missed pulling that forward from my original PR